src/quicr_client.cpp

```diff
@@ -1,665 +1,113 @@
-                  "Received malformed message with unknown fatal error");
-          throw;
-      }
-    }
-  }
-
-private:
-  QuicRClient& client;
-};
-
-void
-QuicRClient::make_transport(RelayInfo& relay_info,
-                            qtransport::TransportConfig tconfig,
-                            qtransport::LogHandler& logger)
-{
-  qtransport::TransportRemote server = {
-    .host_or_ip = relay_info.hostname,
-    .port = relay_info.port,
-    .proto = relay_info.proto == RelayInfo::Protocol::UDP
-               ? qtransport::TransportProtocol::UDP
-               : qtransport::TransportProtocol::QUIC,
-  };
-  transport_delegate = std::make_unique<QuicRTransportDelegate>(*this);
-
-
-  transport = qtransport::ITransport::make_client_transport(
-    server, std::move(tconfig), *transport_delegate, logger);
-
-  transport_context_id = transport->start();
-
-  while (transport->status() == qtransport::TransportStatus::Connecting) {
-    std::ostringstream log_msg;
-    log_msg << "Waiting for client to be ready, got: " << int(transport->status());
-    logger.log(qtransport::LogLevel::info, log_msg.str());
-    std::this_thread::sleep_for(std::chrono::milliseconds(100));
-  }
-
-  if (transport->status() != qtransport::TransportStatus::Ready) {
-    std::ostringstream log_msg;
-    log_msg << "Transport failed to connect to server, got: "
-            << int(transport->status());
-    logger.log(qtransport::LogLevel::fatal, log_msg.str());
-    throw std::runtime_error(log_msg.str());
-  }
-
-
-  transport_stream_id = transport->createStream(transport_context_id, false);
-}
-
 ///
 /// QuicRClient
 ///
 
 QuicRClient::QuicRClient(RelayInfo& relay_info,
                          qtransport::TransportConfig tconfig,
                          qtransport::LogHandler& logger)
-  : log_handler(logger)
 {
-  log_handler.log(qtransport::LogLevel::info, "Initialize QuicRClient");
-
-  if (relay_info.proto == RelayInfo::Protocol::UDP) {
-    // For plain UDP, pacing is needed. Wtih QUIC it's not needed
-    need_pacing = true;
+  switch(relay_info.proto)
+  {
+    case RelayInfo::Protocol::UDP:
+      [[fallthrough]];
+    case RelayInfo::Protocol::QUIC:
+    client_session =
+      std::make_unique<QuicRClientRawSession>(relay_info, tconfig, logger);
+      break;
+    default:
+      throw QuicRClientException("Unsupported relay protocol");
+      break;
   }
-
-  make_transport(relay_info, std::move(tconfig), logger);
-}
-
-QuicRClient::QuicRClient(std::shared_ptr<ITransport> transport_in)
-  : log_handler(def_log_handler)
-{
-  transport = transport_in;
 }
 
-QuicRClient::~QuicRClient()
+QuicRClient::QuicRClient(std::shared_ptr<qtransport::ITransport> transport_in)
 {
-  removeSubscribeState(true, {},
-                      SubscribeResult::SubscribeStatus::ConnectionClosed);
-  transport.reset(); // wait for transport close
+  client_session =
+    std::make_unique<QuicRClientRawSession>(transport_in);
```

Comment:
So, it's worth noting that QuicrClient has become mostly a wrapper of QuicrSession. THis isn't necessarily a bad thing, but since these 2 constructors are the only real difference, I'd like to make a pitch:

Why not do away with QuicrClient/QuicrServer, and instead turn these into standard library style make methods:
```cpp
std::shared_ptr<quicr::ClientSession> make_client_session(...);
std::shared_ptr<quicr::ServerSession> make_server_session(...);
```